### ISSUE TITLE
Create Alternative Main View (List)

### ISSUE THREAD

**Issue by DavidTPate** [antew/RedditInPictures#1]
Ideally we would have a simple list item which includes the most common functionality easily. Here's an example from "Reddit is Fun". Obviously, we want ours to be focused on images mainly so having the Upvote/Downvote and Count on the left then the Image thumbnail directly next to it followed by the post title and meta content (subreddit, etc) would be ideal.

![image](https://f.cloud.github.com/assets/2223120/1955479/a9031980-81e9-11e3-9019-e552dc969a25.png)

Long Pressing the List Item or swiping it would either expand a menu or switch the item out with a menu providing functionality to Share, Upvote/Downvote, Open in Browser, etc.

**Comment by DavidTPate** [comment]
We probably want to persist their choice of Grid/List as well in Shared Preferences. Thinking that list should be the default as it is more common on the platform.

**Comment by antew** [comment]
I started work on this with 7098e9dd815db2bba4aa1f74feff1f3fbadb2ebf

Here's a preview of how it looks now:
![image](https://f.cloud.github.com/assets/1693421/2073049/359655be-8d4e-11e3-99e0-fcc314c14e0a.png)

I made the change quick and dirty just to get reacquainted with the code.  The ImageGridFragment and the new ImageListFragment share enough code that they should really share a base class.

ImageGridActivity needs to be renamed to something like 'ImageActivity' to indicate that it supports more than just a GridView.

Still need to add more details on the post and work on the layout.  That will have to wait for the next commit though...
